src/main/java/it/pagopa/transactions/utils/NodoOperations.java

```diff
@@ -58,6 +59,42 @@ public NodoOperations(
         this.openTelemetryUtils = openTelemetryUtils;
     }
 
+    public Mono<PaymentRequestInfo> activatePaymentRequest(
+                                                           RptId rptId,
+                                                           IdempotencyKey idempotencyKey,
+                                                           Integer amount,
+                                                           String transactionId,
+                                                           Integer paymentTokenTimeout,
+                                                           String idCart,
+                                                           String dueDate,
+                                                           Transaction.ClientId clientId
+    ) {
+        return activatePaymentRequest(
+                rptId,
+                idempotencyKey,
+                amount,
+                transactionId,
+                paymentTokenTimeout,
+                idCart,
+                dueDate
+        )
+                .flatMap(paymentRequestInfo -> {
+                    if (clientId == Transaction.ClientId.WISP_REDIRECT
+                            && paymentRequestInfo.creditorReferenceId() == null) {
+                        return Mono.error(
+                                new InvalidNodoResponseException(
+                                        String.format(
+                                                "Mandatory creditorReferenceId for client %s is missing",
+                                                clientId.name()
+                                        )
+                                )
+                        );
+                    } else {
+                        return Mono.just(paymentRequestInfo);
+                    }
+                });
+    }
+
     public Mono<PaymentRequestInfo> activatePaymentRequest(
```

Comment:
Can we make this overload private? (If the overload that should be used is the one with `clientId`)

Reply:
Sure, done in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/558/commits/6d10998f3e66aca1c9fe9bb279f794d18a4624ca